File: websockets/wsclient.py
import websocket

def on_message(ws, message):
    print(f"Received: {message}")

def on_error(ws, error):
    print(f"Error: {error}")

def on_close(ws, close_status_code, close_msg):
    print("WebSocket closed")

def on_open(ws):
    print("WebSocket connection opened")
    # You can send an initial message here if you want
    ws.send("Hello Server from Python client!")

if __name__ == "__main__":
    # websocket.enableTrace(True)
    # ws = websocket.WebSocketApp("ws://localhost:3000/ws",
    ws = websocket.WebSocketApp("wss://manoj-websockets.mj.dev.dbos.dev/ws",
                                on_open=on_open,
                                on_message=on_message,
                                on_error=on_error,
                                on_close=on_close)

    ws.run_forever()

File: typescript/websockets/wsclient.js
const WebSocket = require('ws');

function onMessage(message) {
    console.log(`Received: ${message}`);
}

function onError(error) {
    console.error(`Error: ${error.message}`);
}

function onClose(code, reason) {
    console.log(`WebSocket closed with code: ${code}, reason: ${reason}`);
}

function onOpen(ws) {
    console.log("WebSocket connection opened");
    ws.send("Hello Server from TypeScript client!");
}

const wsUrl = process.argv[2] || 'ws://localhost:3000/ws';
console.log(`Connecting to ${wsUrl}`);

const ws = new WebSocket(wsUrl);

ws.on('open', () => onOpen(ws));
ws.on('message', onMessage);
ws.on('error', onError);
ws.on('close', onClose);
